modules/api/src/main/java/org/apache/ignite/lang/MarshallerException.java

```diff
@@ -28,6 +32,18 @@ public class MarshallerException extends IgniteException {
      * @param cause Non-null throwable cause.
      */
     public MarshallerException(Throwable cause) {
-        super(cause);
+        super(Common.INTERNAL_ERR, cause);
+    }
+
+    /**
+     * Creates an exception with the given trace ID, error code, detailed message, and cause.
+     *
+     * @param traceId Unique identifier of the exception.
+     * @param code Full error code.
+     * @param message Detailed message.
+     * @param cause Optional nested exception (can be {@code null}).
+     */
+    public MarshallerException(UUID traceId, int code, String message, @Nullable Throwable cause) {
```

Comment:
This constructor is need to preserve information provided with exception created via `MarshallerException(Throwable cause)`.